rasa/core/brokers/kafka.py

```diff
@@ -216,4 +212,4 @@ def _close(self) -> None:
     @rasa.shared.utils.common.lazy_property
     def rasa_environment(self) -> Optional[Text]:
         """Get value of the `RASA_ENVIRONMENT` environment variable."""
-        return os.environ.get("RASA_ENVIRONMENT")
+        return os.environ.get("RASA_ENVIRONMENT", "RASA_ENVIRONMENT_NOT_SET")
```

Comment:
@melindaloubser1 I've added this default because otherwise kafka publish fails with
```
2021-12-31 16:26:18 ERROR    rasa.core.brokers.kafka  - Could not publish message to kafka url 'localhost'. Failed with error: encoding without a string argument
```

I think we also need to document this?

Reply:
noticed a similar change [here](https://github.com/RasaHQ/rasa/pull/10582/files)

Reply:
Sorry I was not online so just seeing this now - is it resolved/documented already?

Reply:
@melindaloubser1 I've added the default `RASA_ENVIRONMENT_NOT_SET` but I haven't documented it as I'm not sure what the use case for this is. Seeing you added this you may be better suited to document it? We can also create a ticket for someone else. What do you think?

Reply:
Ohh wait a sec I remember this now! The default here should be `production`. This is about multiple deployment environments.

Reply:
I thought the `RASA_ENVIRONMENT` already defaulted to `production` though 🤔? Under which circumstances did you see the error above?

Reply:
Oh I see, is there a document around this? I just got above error while running it locally so no env variables set by me.

Reply:
Context-document is https://rasa.com/docs/rasa-x/enterprise/deployment-environments, but nothing for OSS because the parameter is meaningless without Rasa Enterprise attached. 
The default is set up in Rasa Enterprise, but not Rasa OSS without Enterprise attached, so I was wrong. 
In https://github.com/RasaHQ/rasa/pull/10413, I should have accounted for Kafka running without Enterprise setting the `RASA_ENVIRONMENT` var. I'm wondering if it's more reasonable to disable adding the header if the env var is not set.

Reply:
I'll open a PR to fix it and tag you

Reply:
https://github.com/RasaHQ/rasa/pull/10634

Reply:
and a backport since the original was also backported: https://github.com/RasaHQ/rasa/pull/10635

Reply:
@melindaloubser1 I think there's a typo in  https://rasa.com/docs/rasa-x/enterprise/deployment-environments where environments are listed after `By default, two environments are defined:`. Worker should be `development`. 

I've approved your PRs.

Reply:
The image shows `development` in addition to `worker`, but only `worker` is default, so no typo as far as I can see.

Reply:
I mean there are two environments, production and development but the list shows `worker` as an environment:
```
By default, two environments are defined:

production
worker
```

Reply:
It's referring to the right hand side box:
![image](https://user-images.githubusercontent.com/32034278/148377127-3d2362b7-4e02-468b-9349-80d520ce60b2.png)

`development` is not a default environment. It must be added manually.

Reply:
@melindaloubser1 Hmm, I see. I think I'm just confused then :D I thought prod and dev are default environments whereas worker would be separate instance just for training. Thanks for the explanation.